File: src/main/golang/leetcode/editor/cn/[6]Zigzag Conversion.go
package lc

import "strings"

//leetcode submit region begin(Prohibit modification and deletion)
func convert(s string, numRows int) string {
	if numRows == 1 {
		return s
	}
	var array = make([][]rune, numRows)
	var index int
	var direction = 1
	for _, c := range s {
		array[index] = append(array[index], c)
		index += direction
		if index == 0 {
			direction = 1
		} else if index == numRows-1 {
			direction = -1
		}
	}
	var ans []string
	for _, runes := range array {
		ans = append(ans, string(runes))
	}
	return strings.Join(ans, "")
}

//leetcode submit region end(Prohibit modification and deletion)

File: src/main/golang/leetcode/editor/cn/[6]Zigzag Conversion.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows <= 1: return s
        lst = ["" for _ in range(numRows)]
        curr, direction = 0, -1
        for c in s:
            lst[curr] += c
            if curr == 0 or curr == numRows - 1: direction = -direction
            curr += direction
        return "".join(lst)
# leetcode submit region end(Prohibit modification and deletion)
